src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,173 @@
+        Node<T> node = getNodeByIndex(index);
+        T oldItem = node.item;
+        node.item = value;
+        return oldItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkElementPosition(index);
+        return unlinkNode(getNodeByIndex(index));
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        return unlinkNode(object);
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
+        return size == 0;
+    }
+
+    private void linkNode(T item, Node<T> nearestNode) {
+        Node<T> previousNode = nearestNode.prev;
+        Node<T> newNode = new Node<>(previousNode, item, nearestNode);
+        nearestNode.prev = newNode;
+        if (previousNode != null) {
+            previousNode.next = newNode;
+        } else {
+            first = newNode;
+        }
+        size++;
+    }
+
+    private void linkNode(T item) {
+        Node<T> currentLastNode = last;
+        Node<T> newLastNode = new Node<>(currentLastNode, item, null);
+        last = newLastNode;
+        if (currentLastNode != null) {
+            currentLastNode.next = newLastNode;
+        } else {
+            first = newLastNode;
+        }
+        size++;
+    }
+
+    private T unlinkNode(Node<T> node) {
+        Node<T> previousNode = node.prev;
+        Node<T> nextNode = node.next;
+        if (previousNode != null) {
+            previousNode.next = nextNode;
+            node.prev = null;
+        } else {
+            first = nextNode;
+        }
+        if (nextNode != null) {
+            nextNode.prev = previousNode;
+            node.next = null;
+        } else {
+            last = previousNode;
+        }
+        size--;
+        return node.item;
+    }
+
+    private boolean unlinkNode(T item) {
+        Node<T> currentNode = first;
+        for (int i = 0; i < size; i++) {
+            if (item == currentNode.item || item != null && item.equals(currentNode.item)) {
+                unlinkNode(currentNode);
+                return true;
+            }
+            currentNode = currentNode.next;
+        }
         return false;
     }
+
+    private boolean isIndexPosition(int index) {
+        return index >= 0 && index <= size;
+    }
+
+    private boolean isElementPosition(int index) {
+        return index >= 0 && index < size;
+    }
+
+    private String indexOutOfBoundMessage(int index) {
+        return "Index out of bounds [" + index + "]";
+    }
+
+    private void checkElementPosition(int index) {
+        if (!isElementPosition(index)) {
+            throw new IndexOutOfBoundsException(indexOutOfBoundMessage(index));
+        }
+    }
+
+    private void checkIndexPosition(int index) {
+        if (!isIndexPosition(index)) {
+            throw new IndexOutOfBoundsException(indexOutOfBoundMessage(index));
+        }
+    }
```

Comment:
can we create one method instead of two?
for example, we can get two or more parameters in a method if needed